qt/applications/workbench/workbench/utils/gather_interfaces.py

```diff
@@ -0,0 +1,85 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2023 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+#  This file is part of the mantid workbench.
+
+"""
+Helper functions to collect the names of interfaces
+"""
+
+import os
+from typing import AnyStr, List, Dict
+
+from mantid.kernel import ConfigService, logger
+from mantidqt.usersubwindowfactory import UserSubWindowFactory
+
+
+def get_interface_dir() -> AnyStr:
+    """
+    Returns the path to the directory containing the mantidqt interfaces launch scripts
+    This is the path to the mantidqtinterfaces package
+    """
+    import mantidqtinterfaces
+
+    return os.path.dirname(mantidqtinterfaces.__file__)
+
+
+def _discover_python_interfaces(interface_dir) -> Dict[str, List[str]]:
+    """Return a dictionary mapping a category to a set of named Python interfaces"""
+    items = ConfigService["mantidqt.python_interfaces"].split()
+    # detect the python interfaces
+    interfaces = {}
+    for item in items:
+        key, script_name = item.split("/")
+        if not os.path.exists(os.path.join(interface_dir, script_name)):
+            logger.warning('Failed to find script "{}" in "{}"'.format(script_name, interface_dir))
+            continue
+        interfaces.setdefault(key, []).append(script_name)
+
+    return interfaces
+
+
+def _discover_registers_to_run(interface_dir: str) -> Dict[str, List[str]]:
+    items = ConfigService["mantidqt.python_interfaces"].split()
+    try:
+        register_items = ConfigService["mantidqt.python_interfaces_io_registry"].split()
+    except KeyError:
+        return {}
+
+    registers_to_run = {}
+    for item in items:
+        key, script_name = item.split("/")
+        reg_name = script_name[:-3] + "_register.py"
+        if reg_name in register_items and os.path.exists(os.path.join(interface_dir, reg_name)):
+            registers_to_run.setdefault(key, []).append(reg_name)
+
+    return registers_to_run
+
+
+def _discover_cpp_interfaces(interfaces: Dict[str, List[str]]):
```

Comment:
I have a similar comment here to the one above. It reminds me of this line in the `CppInterfacesStartupTest`
https://github.com/mantidproject/mantid/blob/c1daec527052e9b88866bece07e6fadf35d6030c/Testing/SystemTests/tests/qt/CppInterfacesStartupTest.py#L31

Might there be a way to simplify this function? And would it be possible to use your utility function in the `CppInterfacesStartupTest`?